### DISCUSSION TITLE
Solax X1-Hybrid Gen3 Modbus integration instal failure

### DISCUSSION THREAD

**Discussion by romaning82** [wills106/homeassistant-solax-modbus#941] [Q&A]
639 / 5 000
Hello, I would like to ask for your advice. I am trying to install Modbus integration for Solax.
Solax Inverter X1-Hybrid Gen3 (I think (H1E37...)
HA Raspberry 3B+
I connected the HA via a USB RS485 converter (meterA,meterB)
I tried:
modbus:
 - name: solax
 type: serial
 method: lip
 port: /dev/ttyUSB0
 Baud rate: 9600
 stop bits: 1
 bytesize: 8
 parity: N
 timeout: 1
 sensors:
 - name: Solax Voltage
 unit_of_measurement: V
 slaves: 1
 address: 0
 input_type: input

I am getting info from the sensor.

But when I want to install the integration it ends up with an error
see the attached log
[home-assistant_2024-06-26T13-36-12.758Z.log](https://github.com/user-attachments/files/15990800/home-assistant_2024-06-26T13-36-12.758Z.log)
Please, give me an advice
Thank you

**Comment by romaning82** [comment]
Sorry 
I found this site
https://github.com/wills106/homeassistant-solax-modbus/discussions/520  
please detete

**Comment by chaosnature** [comment]
Hi,

can I get assistance by any luck 
I am trying to integrate my  sofar_hyd3k-6k-es with this project code, but i get no Device or Entitie found after following the process through.

my datalogger is LSW3_15_FFFF_1.0.96

and thanks for the good hard work put in.

regards
